### ISSUE TITLE
Trailing buy/sell or sticky order buy/sell

### ISSUE THREAD

**Issue by ingnelson** [Effanuel/BitMEX-scaled-orders#10]
Hello.
I really like your software, I use it .
Could you add a Trailing buy/sell order or sticky order buy/sell order to buy or sell faster and Chase the price if it moves as close as possible and excecute it automatically, please .

**Comment by Effanuel** [comment]
Hey, this kind of feature has been on my mind for a while. Since I have university and work stuff right now, I cannot assure you that will be done anytime soon. *(maybe 2-3 months from now)*

The process of order trailing using websocket is more difficult than it sounds.
I am somewhat working on the [best-price](https://github.com/Effanuel/BitMEX-scaled-orders/tree/best-price) which is basically the thing you are talking about.

*Note: Thanks for using the tool.*

**Comment by ingnelson** [comment]
@Effanuel maybe a code from other project can help a little : 
https://github.com/BigBitsIO/BitMEXAssistant/blob/master/BitMEXAssistant/BitMEXAssistant/Bot.cs

The option:  limitnow

But you do it better

**Comment by Effanuel** [comment]
@ingnelson I almost every piece I need for trailing price, but the problem is setting it all up, testing and making sure everything is handled correctly, so there are no random entire balance limit orders or something similar.

The process of trailing price is this:
* Place post-only order depending on the current price of the symbol on either askPrice or bidPrice, based on side; 
* Check if order was added to the order book;
* If it was added:
   * Send subscribe to websocket to get current open orders;
   * Check for changes in price, or if order got filled;
   * if price changes,
      * amend the order *(currently this part is missing)*
      * if order amending failed, *(currently this part is missing)*
         * kill open orders or retry *(currently this part is missing)*
   * if order got filled.
      * unsubscribe from websocket, clear state, reset, *(currently this part is missing)*
* if order didn't get added, send error or a warning depending if post-only got triggered or not

So the repo you linked isn't really going to help since it's just a matter of time of setting it all up.

**Comment by ingnelson** [comment]
Got it , I just wait for your great works. 
Cheer up !!!

**Comment by ingnelson** [comment]
@Effanuel best-price option is ready to use ?

**Comment by Effanuel** [comment]
@ingnelson Hey, I am planning to release it by the end of the next week. If you want to try it, you can do `git checkout trailing` and build from there. It's in a sort of development faze, so the code  may not be very clean etc. The most notable thing is that config keys go into `./client/.env` file instead of `./api/.env`

**Comment by Poilaucul** [comment]
Tried the best-price, thi is what I get:

![image](https://user-images.githubusercontent.com/2638123/90346234-aa13eb80-e027-11ea-9f17-6bb787c0bd1e.png)

```
 ll BitMEX-scaled-orders/client/build/
total 48
drwxr-xr-x 2 root root 4096 août  16 04:35 ./
drwxr-xr-x 7 root root 4096 août  16 04:35 ../
-rw-r--r-- 1 root root 9662 août  16 04:35 favicon.ico
-rw-r--r-- 1 root root 5347 août  16 04:35 logo192.png
-rw-r--r-- 1 root root 9664 août  16 04:35 logo512.png
-rw-r--r-- 1 root root  503 août  16 04:35 manifest.json
-rw-r--r-- 1 root root   57 août  16 04:35 robots.txt

```

**Comment by Effanuel** [comment]
@Poilaucul The branch is `trailing` not `best-price`

**Comment by Poilaucul** [comment]
My bad, we have two subjects mixed in this thread so I figured i would post
about the one I'm only interested in; best-price feature.
So trailing branch is ahead of the best-price one regarding best-price
matching?

>

**Comment by Effanuel** [comment]
Yes,  at first I made the `best-price` branch, but the name, however accurate it is, is misleading, so I changed it to `trailing`.
The feature that is there is basically placing a limit order at the "best" price *(closest to market  price)*, which moves if not filled based on the market price.

**Comment by Effanuel** [comment]
This feature will be key in implementing more advanced features such as moving stop-loss to breakeven if some target was hit.

**Comment by Effanuel** [comment]
I will update the list of the features I want to add, but I don't always have the time these days, since I am working and going to uni. @Poilaucul

**Comment by Poilaucul** [comment]
Got the same `{"error":"ENOENT: no such file or directory, stat '/home/desktop/BitMEX-scaled-orders/client/build/index.html'"}` with the trailing branch.

**Comment by Effanuel** [comment]
Did you make sure you made a refresh reinstall of node modules?
* git clone ...
* cd api
* npm run init:packages
* npm run build
* npm run prod

If that doesn't work, look into the build folder and search for **index.html** file. This might have something to do with you working on linux or your folder permissions

**Comment by Poilaucul** [comment]
Works now, thanks for the tip.

**Comment by Effanuel** [comment]
Updated `trailing` branch with some fixes.
